src/Sulu/Bundle/DocumentManagerBundle/Command/PhpCRCleanerCommand.php

```diff
@@ -0,0 +1,244 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\DocumentManagerBundle\Command;
+
+use PHPCR\PathNotFoundException;
+use PHPCR\SessionInterface;
+use Sulu\Component\Content\Metadata\BlockMetadata;
+use Sulu\Component\Content\Metadata\ComponentMetadata;
+use Sulu\Component\Content\Metadata\Factory\Exception\DocumentTypeNotFoundException;
+use Sulu\Component\Content\Metadata\Factory\Exception\StructureTypeNotFoundException;
+use Sulu\Component\Content\Metadata\Factory\StructureMetadataFactoryInterface;
+use Sulu\Component\Content\Metadata\ItemMetadata;
+use Sulu\Component\Content\Metadata\PropertyMetadata;
+use Sulu\Component\Content\Metadata\StructureMetadata;
+use Sulu\Component\PHPCR\PropertyParser\Property;
+use Sulu\Component\PHPCR\PropertyParser\PropertyParserInterface;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Webmozart\Assert\Assert;
+
+class PhpCRCleanerCommand extends Command
+{
+    protected static $defaultName = 'sulu:document:clean';
+
+    private OutputInterface $output;
+
+    /** @var array<mixed> */
+    private array $nodesToDelete = [];
+
+    public function __construct(
+        private PropertyParserInterface $propertyParser,
+        private SessionInterface $session,
+        private StructureMetadataFactoryInterface $structureMetaDataFactory
+    ) {
+        parent::__construct();
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $this->output = $output;
+        $queryManager = $this->session->getWorkspace()->getQueryManager();
+        $rows = $queryManager->createQuery('SELECT * FROM [nt:unstructured]', 'JCR-SQL2')->execute();
```

Comment:
Maybe could filter for pages here as well, as this thing does not support snippets and reduce the number of errors later.